syntax = "proto3";

package tensorflow.nufft;

enum FftwPlanningRigor {
  AUTO = 0;
  ESTIMATE = 1;
  MEASURE = 2;
  PATIENT = 3;
  EXHAUSTIVE = 4;
}

enum PointBounds {
  STRICT = 0;
  EXTENDED = 1;
  INFINITE = 2;
}

message FftwOptions {
  FftwPlanningRigor planning_rigor = 1;
}

message DebuggingOptions {
  bool check_bounds = 1;
}

message Options {
  DebuggingOptions debugging = 1;
  FftwOptions fftw = 2;
  int32 max_batch_size = 3;
  PointBounds point_bounds = 4;
}
